File: Entrada de Dados/AlfaNum/alfanum.go
package main

import (
  "fmt"
  "strings"
)

func VALIDA_HEXA(VALOR string) bool {
  for _, C := range VALOR {
    CARACTERE := strings.ToUpper(string(C))
    if !((CARACTERE >= "0" && CARACTERE <= "9") || (CARACTERE >= "A" && CARACTERE <= "F")) {
      return false
    }
  }
  return true
}

func main() {
  fmt.Println("")
  var VALOR_HEX string
  SAIR_LOOP := false
  for !SAIR_LOOP {
    fmt.Print("Entre valor hexadecimal de 4 caracteres: ")
    fmt.Scanln(&VALOR_HEX)
    VALOR_HEX = strings.TrimSpace(VALOR_HEX)
    if len(VALOR_HEX) != 4 {
      fmt.Println("Valor invalido, deve ter exatamente 4 caracteres.\n")
    } else if !VALIDA_HEXA(VALOR_HEX) {
      fmt.Println("Valor invalido, apenas caracteres [0-9] e [A-F].\n")
    } else {
      SAIR_LOOP = true
    }
  }
  VALOR_HEX_CONVERTIDO := ""
  for _, C := range VALOR_HEX {
    CARACTERE := strings.ToUpper(string(C))
    VALOR_HEX_CONVERTIDO += CARACTERE
  }
  fmt.Println("Valor fornecido: ", VALOR_HEX_CONVERTIDO, "\n")
}

File: Entrada de Dados/AlfaNum/alfanum.cpp
#include <iostream>
#include <cstring>
#include <cctype>

using namespace std;

bool VALIDA_HEXA(const string& VALOR) {
  for (char CARACTERE : VALOR) {
    char CARACTERE_MAISCULO = toupper(CARACTERE);
    if (!(isdigit(CARACTERE_MAISCULO) || (CARACTERE_MAISCULO >= 'A' && CARACTERE_MAISCULO <= 'F')))
      return false;
  }
  return true;
}

string VALOR_HEX;
  
int main() {
  cout << endl;
  do {
    cout << "Entre valor hexadecimal de 4 caracteres: ";
    getline(cin, VALOR_HEX);
    if (VALOR_HEX.length() != 4) {
      cout << "Valor invalido, deve ter exatamente 4 caracteres." << endl << endl;
      continue;
    }
    if (!VALIDA_HEXA(VALOR_HEX)) {
      cout << "Valor invAlido, apenas caracteres [0-9] e [A-F]." << endl << endl;
      continue;
    }
    break;
  } while (true);
  for (char& c : VALOR_HEX) {
    c = toupper(c);
  }
  cout << "Valor fornecido: " << VALOR_HEX << endl << endl;
  return 0;
}
